bcm63xx: switch to Kernel 5.10

aparcar opened the pull request:
CI build with artifacts: https://github.com/aparcar/openwrt/actions/runs/1289780656

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar committed:
bcm63xx: switch to Kernel 5.10

Bump the last missing target to Kernel 5.10. While this requires a work
around to boot it will allow more people to test the new Kernel before
the upcomming release.

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar commented:
Anyone interested in this target?

Noltari commented:
> Anyone interested in this target?

Hi @aparcar, I'm interested but for personal reasons I don't have much time to test this properly.
Anyhow, if I remember correctly, there was an issue with some NANDs not being correctly detected with the kernel update.
@danitool can probably help me here since I think it affected the Sercomm H500-s...

aparcar commented:
@Noltari @danitool any updates on this?

danitool commented:
> @Noltari @danitool any updates on this?

Hi @aparcar, the problem with the NAND flash in Macronix chips is is still there. Current snapshot failure:
```
[    0.000000] Linux version 5.10.87 (dani@tool) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 11.2.0 r18394-962c585580) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Tue Dec 21 23:44:45 2021
[    0.000000] Detected Broadcom 0x63268 CPU revision d0
[    0.000000] CPU frequency is 400 MHz
[    0.000000] 128MB of RAM installed
[    0.000000] board_bcm963xx: Boot address 0xb8000000
[    0.000000] board_bcm963xx: CFE version: 1.0.38-118.3
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0002a080 (Broadcom BMIPS4350)
[    0.000000] board: board name: BXK00C-1.6
[    0.000000] MIPS: machine is Sercomm H500-s vfes
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 16 bytes.
[    0.000000] Primary data cache 32kB, 2-way, VIPT, cache aliases, linesize 16 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 14 pages/cpu s27344 r8192 d21808 u57344
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: rootfstype=squashfs,ubifs noinitrd console=ttyS0,115200
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 113204K/131072K available (6424K kernel code, 633K rwdata, 1316K rodata, 7876K init, 211K bss, 17868K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] random: get_random_bytes called from start_kernel+0x3d0/0x600 with crng_init=0
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302233 ns
[    0.000015] sched_clock: 32 bits at 200MHz, resolution 5ns, wraps every 10737418237ns
[    0.008153] Calibrating delay loop... 398.13 BogoMIPS (lpj=1990656)
[    0.074573] pid_max: default: 32768 minimum: 301
[    0.079680] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.087196] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.100196] rcu: Hierarchical SRCU implementation.
[    0.105799] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.114217] smp: Bringing up secondary CPUs ...
[    0.120402] SMP: Booting CPU1...
[    4.572693] Primary instruction cache 64kB, VIPT, 4-way, linesize 16 bytes.
[    4.572717] Primary data cache 32kB, 2-way, VIPT, cache aliases, linesize 16 bytes
[    4.572998] CPU1 revision is: 0002a080 (Broadcom BMIPS4350)
[    0.189070] Synchronize counters for CPU 1: 
[    0.209736] SMP: CPU1 is running
[    0.209742] done.
[    0.219643] smp: Brought up 1 node, 2 CPUs
[    0.235720] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.245894] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.253088] pinctrl core: initialized pinctrl subsystem
[    0.259528] NET: Registered protocol family 16
[    0.378252] clocksource: Switched to clocksource MIPS
[    0.387715] NET: Registered protocol family 2
[    0.392628] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.401611] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.410484] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.418472] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.425893] TCP: Hash tables configured (established 1024 bind 1024)
[    0.433150] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.440119] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.448307] NET: Registered protocol family 1
[    0.452832] PCI: CLS 0 bytes, default 16
[    0.483420] random: fast init done
[    0.725833] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.740804] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.746838] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.761746] bcm63268-pinctrl 100000c0.pin-controller: registered at mmio (ptrval)
[    0.771179] 10000180.serial: ttyS0 at MMIO 0x10000180 (irq = 13, base_baud = 1562500) is a bcm63xx_uart
[    0.780972] printk: console [ttyS0] enabled
[    0.780972] printk: console [ttyS0] enabled
[    0.789569] printk: bootconsole [early0] disabled
[    0.789569] printk: bootconsole [early0] disabled
[    0.812291] bcm6368_nand 10000200.nand: there is not valid maps for state default
[    0.823019] nand: device found, Manufacturer ID: 0xc2, Chip ID: 0xf1
[    0.829592] nand: Macronix MX30LF1G18AC
[    0.833525] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.841306] bcm6368_nand 10000200.nand: detected 128MiB total, 128KiB blocks, 2KiB pages, 16B OOB, 8-bit, BCH-4
[    0.857758] Bad block table not found for chip 0
[    0.867999] Bad block table not found for chip 0
[    0.872690] Scanning device for bad blocks
[    0.909739] CPU 0 Unable to handle kernel paging request at virtual address 0040000c, epc == 80064e4c, ra == 80066e30
[    0.920631] Oops[#1]:
[    0.922965] CPU: 0 PID: 0 Comm: swapper/0 Not tainted 5.10.87 #0
[    0.929136] $ 0   : 00000000 00000001 00400000 00000002
[    0.934511] $ 4   : 81438080 8140be2c 00000000 81408300
[    0.939887] $ 8   : 8140a000 00001ff0 00000001 8114af80
[    0.945263] $12   : 00000400 8143bc80 8143bc80 00000000
[    0.950638] $16   : 80ee1f40 00000003 81438080 807b460c
[    0.956014] $20   : 00000002 807a9880 806e9670 806e9648
[    0.961392] $24   : 00000002 806549f8                  
[    0.966767] $28   : 807a2000 8140bde0 80820000 80066e30
[    0.972143] Hi    : 00000008
[    0.975101] Lo    : cccccccf
[    0.978082] epc   : 80064e4c __task_rq_lock+0x38/0xc0
[    0.983266] ra    : 80066e30 try_to_wake_up+0xb4/0x5a4
[    0.988541] Status: 10008f02 KERNEL EXL 
[    0.992574] Cause : 00800008 (ExcCode 02)
[    0.996695] BadVA : 0040000c
[    0.999652] PrId  : 0002a080 (Broadcom BMIPS4350)
[    1.004488] Modules linked in:
[    1.007633] Process swapper/0 (pid: 0, threadinfo=(ptrval), task=(ptrval), tls=00000000)
[    1.015957] Stack : 81438000 00000003 00000000 81438000 81438080 00000003 00000000 10008f00
[    1.024567]         8143852c 80066e30 00000400 0000002e 00000329 00000000 00000000 00000001
[    1.033178]         00000001 00000000 00000000 00000000 81422180 82368cc4 10008f00 8140bf0c
[    1.041789]         0000003a 807a9880 806e9670 806e9648 80820000 80080920 80ee1f40 00000000
[    1.050400]         80ee1f40 82368cc0 82368cc0 80080d90 81421a00 00000000 00000000 00000000
[    1.059010]         ...
[    1.061520] Call Trace:
[    1.064038] [<80064e4c>] __task_rq_lock+0x38/0xc0
[    1.068881] [<80066e30>] try_to_wake_up+0xb4/0x5a4
[    1.073831] [<80080920>] swake_up_locked+0x28/0x58
[    1.078734] [<80080d90>] complete+0x44/0x64
[    1.083065] [<8043b450>] brcmnand_irq+0xa4/0xac
[    1.087719] [<80090a3c>] __handle_irq_event_percpu+0x70/0x1b8
[    1.093607] [<80090c4c>] handle_irq_event+0x50/0xe8
[    1.098624] [<80094fbc>] handle_level_irq+0xf0/0x1fc
[    1.103732] [<80090118>] generic_handle_irq+0x44/0x5c
[    1.108946] [<80396b10>] bcm6345_periph_irq_handle+0x110/0x1c0
[    1.114932] [<80090118>] generic_handle_irq+0x44/0x5c
[    1.120152] [<8065521c>] do_IRQ+0x1c/0x2c
[    1.124254] [<80396fc4>] plat_irq_dispatch+0x60/0xd0
[    1.129371] [<80015bf0>] handle_int+0x150/0x15c
[    1.134015] [<80015a60>] __r4k_wait+0x20/0x40
[    1.138492] 
[    1.140008] Code: 24140002  26101f40  8e420004 <8c42000c> 00021080  02621021  8c510000  02118821  0c19540e 
[    1.150070] 
[    1.151660] ---[ end trace 625caeee62f8fa21 ]---
[    1.156366] Kernel panic - not syncing: Fatal exception in interrupt
[    1.162890] ------------[ cut here ]------------
[    1.167651] WARNING: CPU: 0 PID: 0 at kernel/smp.c:633 smp_call_function_many_cond+0x438/0x454
[    1.176503] Modules linked in:
[    1.179651] CPU: 0 PID: 0 Comm: swapper/0 Tainted: G      D           5.10.87 #0
[    1.187256] Stack : 10008f00 00000000 80820000 8008d84c 80830000 80725d5c 00000000 00000000
[    1.195864]         8140bb24 80ff0000 806f4060 807b74f8 807b6fe7 10008f00 8140bac8 5f0039dd
[    1.204474]         00000000 00000000 806f4060 8140b960 00000000 00000000 ffffefff ffffffea
[    1.213086]         0000008a 8140b96c 0000008a 807c0818 00000000 10008f00 00000000 806f0000
[    1.221696]         00000009 807b7100 00000000 00000000 00000678 00000000 00000000 80ff0000
[    1.230306]         ...
[    1.232817] Call Trace:
[    1.235353] [<8001a210>] show_stack+0x30/0x100
[    1.239907] [<803794b8>] dump_stack+0xa8/0xd8
[    1.244407] [<8003aaa4>] __warn+0xc0/0x12c
[    1.248599] [<8003ab6c>] warn_slowpath_fmt+0x5c/0xac
[    1.253704] [<800c3edc>] smp_call_function_many_cond+0x438/0x454
[    1.259885] [<800c3f3c>] smp_call_function+0x24/0x30
[    1.264996] [<8003a510>] panic+0x114/0x31c
[    1.269206] [<8001a4cc>] die+0x124/0x130
[    1.273257] [<800257a0>] do_page_fault+0x534/0x544
[    1.278172] [<8002a674>] tlb_do_page_fault_0+0x124/0x12c
[    1.283638] [<80066e30>] try_to_wake_up+0xb4/0x5a4
[    1.288549] 
[    1.290076] ---[ end trace 625caeee62f8fa22 ]---
[    1.294823] ------------[ cut here ]------------
[    1.299582] WARNING: CPU: 0 PID: 0 at kernel/smp.c:498 smp_call_function_single+0x1f4/0x1fc
[    1.308170] Modules linked in:
[    1.311315] CPU: 0 PID: 0 Comm: swapper/0 Tainted: G      D W         5.10.87 #0
[    1.318921] Stack : 10008f00 00000001 80820000 8008d84c 80830000 80725d5c 00000000 00000000
[    1.327532]         8140bb34 80ff0000 806f4060 807b74f8 807b6fe7 10008f00 8140bad8 5f0039dd
[    1.336142]         00000000 00000000 806f4060 8140b970 00000000 00000000 ffffefff ffffffea
[    1.344753]         000000a2 8140b97c 000000a2 807c0818 00000000 10008f00 00000000 806f0000
[    1.353363]         00000009 807b7100 00000000 00000001 00000798 00000000 00000000 80ff0000
[    1.361974]         ...
[    1.364482] Call Trace:
[    1.367003] [<8001a210>] show_stack+0x30/0x100
[    1.371573] [<803794b8>] dump_stack+0xa8/0xd8
[    1.376056] [<8003aaa4>] __warn+0xc0/0x12c
[    1.380265] [<8003ab6c>] warn_slowpath_fmt+0x5c/0xac
[    1.385373] [<800c3924>] smp_call_function_single+0x1f4/0x1fc
[    1.391284] [<800c3f3c>] smp_call_function+0x24/0x30
[    1.396394] [<8003a510>] panic+0x114/0x31c
[    1.400604] [<8001a4cc>] die+0x124/0x130
[    1.404637] [<800257a0>] do_page_fault+0x534/0x544
[    1.409566] [<8002a674>] tlb_do_page_fault_0+0x124/0x12c
[    1.415029] [<80066e30>] try_to_wake_up+0xb4/0x5a4
[    1.419948] 
[    1.421472] ---[ end trace 625caeee62f8fa23 ]---
[    1.426235] Rebooting in 1 seconds..
�HELO2.425330] triggering watchdog soft-reset...
CPUI
L1CI
```
This failure can be fixed with this simple-ugly patch
```diff
--- a/drivers/mtd/nand/raw/nand_macronix.c
+++ b/drivers/mtd/nand/raw/nand_macronix.c
@@ -323,7 +323,7 @@
 
 	macronix_nand_fix_broken_get_timings(chip);
 	macronix_nand_onfi_init(chip);
-	macronix_nand_block_protection_support(chip);
+	//macronix_nand_block_protection_support(chip);
 	macronix_nand_deep_power_down_support(chip);
 
 	return 0;
```

Regards

aparcar commented:
Well it wouldn't be the first time we add a patch to the Linux kernel, I'm not sure how _ugly_ this patch is. Could you please elaborate.

danitool commented:
The patch disables the macronix_nand_block_protection_support added at some point before 5.10. It's harmless to disable it. It's not present in 5.4 and therefore we can live without it in 5.10 as well.

The ugly part is we're compiling non used functions.

hauke commented:
Did someone try to backport these commits from mainline Linux:
https://github.com/torvalds/linux/commit/22ca05b82d3e3abc2b116a11ee41b6b692b95530
https://github.com/torvalds/linux/commit/a071912636cc3420f54e2a6312c1625ac763cf03

hauke commented:
This patch is already in the stable kernel and should get removed:
target/linux/bcm63xx/patches-5.10/434-nand-brcmnand-fix-OOB-R-W-with-Hamming-ECC.patch

danitool commented:
> Did someone try to backport these commits from mainline Linux: [torvalds/linux@22ca05b](https://github.com/torvalds/linux/commit/22ca05b82d3e3abc2b116a11ee41b6b692b95530) [torvalds/linux@a071912](https://github.com/torvalds/linux/commit/a071912636cc3420f54e2a6312c1625ac763cf03)

Hi @hauke, I've tested the patches without success. With them the error is a bit different:
```
[    0.000000] Linux version 5.10.96 (dani@tool) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 11.2.0 r18729-8c1d5129ee) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Fri Feb 4 11:51:48 2022
[    0.000000] Detected Broadcom 0x63268 CPU revision d0
[    0.000000] CPU frequency is 400 MHz
[    0.000000] 128MB of RAM installed
[    0.000000] board_bcm963xx: Boot address 0xb8000000
[    0.000000] board_bcm963xx: CFE version: 1.0.38-118.3
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0002a080 (Broadcom BMIPS4350)
[    0.000000] board: board name: BXK00C-1.6
[    0.000000] MIPS: machine is Sercomm H500-s vfes
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 16 bytes.
[    0.000000] Primary data cache 32kB, 2-way, VIPT, cache aliases, linesize 16 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 14 pages/cpu s27344 r8192 d21808 u57344
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: rootfstype=squashfs,ubifs noinitrd console=ttyS0,115200
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 112116K/131072K available (6433K kernel code, 632K rwdata, 1316K rodata, 8956K init, 211K bss, 18956K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] random: get_random_bytes called from start_kernel+0x3d0/0x600 with crng_init=0
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302233 ns
[    0.000015] sched_clock: 32 bits at 200MHz, resolution 5ns, wraps every 10737418237ns
[    0.008152] Calibrating delay loop... 398.13 BogoMIPS (lpj=1990656)
[    0.074573] pid_max: default: 32768 minimum: 301
[    0.079686] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.087196] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.100175] rcu: Hierarchical SRCU implementation.
[    0.105783] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.114208] smp: Bringing up secondary CPUs ...
[    0.120409] SMP: Booting CPU1...
[    8.071090] Primary instruction cache 64kB, VIPT, 4-way, linesize 16 bytes.
[    8.071113] Primary data cache 32kB, 2-way, VIPT, cache aliases, linesize 16 bytes
[    8.071390] CPU1 revision is: 0002a080 (Broadcom BMIPS4350)
[    0.189078] Synchronize counters for CPU 1: 
[    0.209747] SMP: CPU1 is running
[    0.209753] done.
[    0.219659] smp: Brought up 1 node, 2 CPUs
[    0.235705] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.245876] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.253070] pinctrl core: initialized pinctrl subsystem
[    0.259495] NET: Registered protocol family 16
[    0.373952] clocksource: Switched to clocksource MIPS
[    0.382551] NET: Registered protocol family 2
[    0.387787] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.396616] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.405532] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.413521] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.420878] TCP: Hash tables configured (established 1024 bind 1024)
[    0.428300] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.435221] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.443490] NET: Registered protocol family 1
[    0.448040] PCI: CLS 0 bytes, default 16
[    0.513260] random: fast init done
[    0.750625] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.765612] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.771563] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.786711] bcm63268-pinctrl 100000c0.pin-controller: registered at mmio (ptrval)
[    0.796091] 10000180.serial: ttyS0 at MMIO 0x10000180 (irq = 13, base_baud = 1562500) is a bcm63xx_uart
[    0.805882] printk: console [ttyS0] enabled
[    0.805882] printk: console [ttyS0] enabled
[    0.814485] printk: bootconsole [early0] disabled
[    0.814485] printk: bootconsole [early0] disabled
[    0.837168] bcm6368_nand 10000200.nand: there is not valid maps for state default
[    0.847913] nand: device found, Manufacturer ID: 0xc2, Chip ID: 0xf1
[    0.854468] nand: Macronix MX30LF1G18AC
[    0.858354] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.866248] bcm6368_nand 10000200.nand: detected 128MiB total, 128KiB blocks, 2KiB pages, 16B OOB, 8-bit, BCH-4
[    0.882532] Bad block table not found for chip 0
[    0.892833] Bad block table not found for chip 0
[    0.897592] Scanning device for bad blocks
[    0.949664] ------------[ cut here ]------------
[    0.954465] WARNING: CPU: 0 PID: 0 at kernel/rcu/tree.c:718 rcu_nmi_exit+0x138
```
The kernel log stops there, and there is no reboot.

If I also apply the quick fix:
```diff
--- a/drivers/mtd/nand/raw/nand_macronix.c
+++ b/drivers/mtd/nand/raw/nand_macronix.c
@@ -323,7 +323,7 @@
 
 	macronix_nand_fix_broken_get_timings(chip);
 	macronix_nand_onfi_init(chip);
-	macronix_nand_block_protection_support(chip);
+	//macronix_nand_block_protection_support(chip);
 	macronix_nand_deep_power_down_support(chip);
 
 	return 0;
```
The problem is gone.

Regards

ffainelli commented:
Can you boot with "maxcpus=1" or "nosmp"? I wonder if we have a bad concurrency problem here that is somewhat not visible on other platforms. Thanks!

danitool commented:
With "nosmp " the failure is still there:

```
[    0.000000] Linux version 5.10.96 (dani@tool) (mips-openwrt-linux-musl-gcc (OpenWrt GCC 11.2.0 r18729-8c1d5129ee) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Fri Feb 4 11:51:48 2022
[    0.000000] Detected Broadcom 0x63268 CPU revision d0
[    0.000000] CPU frequency is 400 MHz
[    0.000000] 128MB of RAM installed
[    0.000000] board_bcm963xx: Boot address 0xb8000000
[    0.000000] board_bcm963xx: CFE version: 1.0.38-118.3
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0002a080 (Broadcom BMIPS4350)
[    0.000000] board: board name: BXK00C-1.6
[    0.000000] MIPS: machine is Sercomm H500-s vfes
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 16 bytes.
[    0.000000] Primary data cache 32kB, 2-way, VIPT, cache aliases, linesize 16 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 14 pages/cpu s27344 r8192 d21808 u57344
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: rootfstype=squashfs,ubifs noinitrd console=ttyS0,115200 nosmp
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 112116K/131072K available (6432K kernel code, 633K rwdata, 1316K rodata, 8956K init, 211K bss, 18956K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] random: get_random_bytes called from start_kernel+0x3d0/0x600 with crng_init=0
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302233 ns
[    0.000015] sched_clock: 32 bits at 200MHz, resolution 5ns, wraps every 10737418237ns
[    0.008153] Calibrating delay loop... 398.13 BogoMIPS (lpj=1990656)
[    0.074572] pid_max: default: 32768 minimum: 301
[    0.079685] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.087196] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.100172] rcu: Hierarchical SRCU implementation.
[    0.105781] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.114205] smp: Bringing up secondary CPUs ...
[    0.118891] smp: Brought up 1 node, 1 CPU
[    0.134756] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.144926] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.152121] pinctrl core: initialized pinctrl subsystem
[    0.158528] NET: Registered protocol family 16
[    0.294534] clocksource: Switched to clocksource MIPS
[    0.302344] NET: Registered protocol family 2
[    0.307148] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.315959] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.324712] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.332642] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.339939] TCP: Hash tables configured (established 1024 bind 1024)
[    0.346948] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.353789] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.361384] NET: Registered protocol family 1
[    0.365937] PCI: CLS 0 bytes, default 16
[    0.682902] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.697862] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.703901] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.720623] bcm63268-pinctrl 100000c0.pin-controller: registered at mmio (ptrval)
[    0.729453] 10000180.serial: ttyS0 at MMIO 0x10000180 (irq = 13, base_baud = 1562500) is a bcm63xx_uart
[    0.739228] printk: console [ttyS0] enabled
[    0.739228] printk: console [ttyS0] enabled
[    0.747807] printk: bootconsole [early0] disabled
[    0.747807] printk: bootconsole [early0] disabled
[    0.776326] bcm6368_nand 10000200.nand: there is not valid maps for state default
[    0.789140] nand: device found, Manufacturer ID: 0xc2, Chip ID: 0xf1
[    0.795790] nand: Macronix MX30LF1G18AC
[    0.799683] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    0.807552] bcm6368_nand 10000200.nand: detected 128MiB total, 128KiB blocks, 2KiB pages, 16B OOB, 8-bit, BCH-4
[    0.824172] Bad block table not found for chip 0
[    0.834235] Bad block table not found for chip 0
[    0.838932] Scanning device for bad blocks
[    0.862740] random: fast init done
[    0.879325] CPU 0 Unable to handle kernel paging request at virtual address 0040000c, epc == 8006719c, ra == 80080cec
[    0.890211] Oops[#1]:
[    0.892547] CPU: 0 PID: 0 Comm:  Not tainted 5.10.96 #0
[    0.897911] $ 0   : 00000000 00000001 00000001 00400000
[    0.903287] $ 4   : 81438080 00000003 00000000 81408300
[    0.908662] $ 8   : 8140a000 00001ff0 807b660c 00000d3e
[    0.914039] $12   : 00067d95 12200000 00000000 00000000
[    0.919414] $16   : 81438080 00000003 10008f00 8140bf0c
[    0.924790] $20   : 8143852c 807ab880 806eb8c0 806eb898
[    0.930166] $24   : 01e2aeff 80656c48                  
[    0.935542] $28   : 81422000 8140be08 80820000 80080cec
[    0.940919] Hi    : 00000000
[    0.943875] Lo    : 00000000
[    0.946861] epc   : 8006719c try_to_wake_up+0x27c/0x5a4
[    0.952242] ra    : 80080cec swake_up_locked+0x28/0x58
[    0.957495] Status: 10008f02	KERNEL EXL 
[    0.961529] Cause : 00800008 (ExcCode 02)
[    0.965649] BadVA : 0040000c
[    0.968608] PrId  : 0002a080 (Broadcom BMIPS4350)
[    0.973442] Modules linked in:
[    0.976587] Process  (pid: 0, threadinfo=(ptrval), task=(ptrval), tls=00000001)
[    0.984106] Stack : 32aa0f3c 800be004 00000000 00000000 807351c4 00000000 00000000 00989680
[    0.992706]         81255140 33428f00 81422180 8247c8c4 10008f00 8140bf0c 0000003a 807ab880
[    1.001308]         806eb8c0 806eb898 80820000 80080cec 81130000 00000000 00000000 8247c8c0
[    1.009910]         8247c8c0 8008115c 7fffffff ffffffff 10000f00 800ab334 8247c880 0000003a
[    1.018511]         00000000 8043b910 33428f00 00000000 10000f00 800ac5f8 80b5e458 82485e00
[    1.027112]         ...
[    1.029623] Call Trace:
[    1.032142] [<8006719c>] try_to_wake_up+0x27c/0x5a4
[    1.037163] [<80080cec>] swake_up_locked+0x28/0x58
[    1.042090] [<8008115c>] complete+0x44/0x64
[    1.046419] [<8043b910>] brcmnand_irq+0xa4/0xac
[    1.051074] [<80090e08>] __handle_irq_event_percpu+0x70/0x1b8
[    1.056963] [<80091018>] handle_irq_event+0x50/0xe8
[    1.061978] [<80095388>] handle_level_irq+0xf0/0x1fc
[    1.067087] [<800904e4>] generic_handle_irq+0x44/0x5c
[    1.072302] [<80396ed0>] bcm6345_periph_irq_handle+0x110/0x1c0
[    1.078288] [<800904e4>] generic_handle_irq+0x44/0x5c
[    1.083511] [<8065746c>] do_IRQ+0x1c/0x2c
[    1.087609] [<80397384>] plat_irq_dispatch+0x60/0xd0
[    1.092726] [<80015c10>] handle_int+0x150/0x15c
[    1.097360] 
[    1.098882] Code: ac800000  8e030004  24020001 <8c63000c> 8fbf004c  8fbe0048  8fb70044  8fb60040  8fb5003c 
[    1.108944] 
[    1.110519] CPU 0 Unable to handle kernel paging request at virtual address 00000108, epc == 8006b5a8, ra == 8006b6d4
[    1.121402] Oops[#2]:
[    1.123740] CPU: 0 PID: 0 Comm:  Tainted: G      D           5.10.96 #0
[    1.130539] $ 0   : 00000000 10000f00 00000000 00000000
[    1.135915] $ 4   : 81da7680 00010000 00000000 00989680
[    1.141291] $ 8   : 00989680 00000000 00989680 00000000
[    1.146668] $12   : 00000001 00000000 ffffffe2 00000000
[    1.152043] $16   : 81438080 807b0000 00000000 8140bc18
[    1.157419] $20   : 81130000 81130000 00000000 807b0000
[    1.162795] $24   : 1dcd6500 3b9aca00                  
[    1.168171] $28   : 81422000 8140b9d0 00989680 8006b6d4
[    1.173547] Hi    : 00000000
[    1.176504] Lo    : 00000000
[    1.179490] epc   : 8006b5a8 account_system_index_time+0x44/0x110
[    1.185743] ra    : 8006b6d4 account_system_time+0x60/0xa8
[    1.191378] Status: 10000f02	KERNEL EXL 
[    1.195412] Cause : 00800008 (ExcCode 02)
[    1.199532] BadVA : 00000108
[    1.202490] PrId  : 0002a080 (Broadcom BMIPS4350)
[    1.207326] Modules linked in:
[    1.210469] Process  (pid: 0, threadinfo=(ptrval), task=(ptrval), tls=00000001)
[    1.217989] Stack : 81130000 10000f00 81130000 81128b40 00000010 807b660c 807b0000 00000000
[    1.226590]         8140bc18 8006b6d4 09000006 100affff 09000006 5535da56 00000004 8140baf3
[    1.235191]         81259f88 800aac9c 00000000 8140bc18 81130000 81130000 81259f88 40e208ed
[    1.243793]         00000000 800bdfd4 807b0000 811287f8 09000006 100affff 80720000 80727f94
[    1.252394]         80830000 807b0000 81259d40 81259f88 800bdf50 00000000 81259d00 10000f00
[    1.260995]         ...
[    1.263506] Call Trace:
[    1.266025] [<8006b5a8>] account_system_index_time+0x44/0x110
[    1.271941] [<8006b6d4>] account_system_time+0x60/0xa8
[    1.277242] [<800aac9c>] update_process_times+0x80/0x120
[    1.282720] [<800bdfd4>] tick_sched_timer+0x84/0x2d0
[    1.287802] [<800ab64c>] __hrtimer_run_queues+0x15c/0x234
[    1.293357] [<800ac5b4>] hrtimer_interrupt+0x174/0x398
[    1.298642] [<8001fec4>] c0_compare_interrupt+0x8c/0xd0
[    1.304035] [<80090e08>] __handle_irq_event_percpu+0x70/0x1b8
[    1.309931] [<80090f70>] handle_irq_event_percpu+0x20/0x78
[    1.315575] [<800966e4>] handle_percpu_irq+0x84/0xb4
[    1.320681] [<800904e4>] generic_handle_irq+0x44/0x5c
[    1.325897] [<8065746c>] do_IRQ+0x1c/0x2c
[    1.330013] [<80397384>] plat_irq_dispatch+0x60/0xd0
[    1.335119] [<80015c10>] handle_int+0x150/0x15c
[    1.339783] [<8001a488>] die+0xc0/0x130
[    1.343724] [<800257c0>] do_page_fault+0x534/0x544
[    1.348643] [<8002a694>] tlb_do_page_fault_0+0x124/0x12c
[    1.354126] [<80080cec>] swake_up_locked+0x28/0x58
[    1.359022] 
[    1.360541] Code: 8e060428  ae020368  ae04036c <8cc20108> 8fb30038  10400009  00e08825  8e02042c  10400006 
[    1.370593] 
[    1.372134] ---[ end trace a5823539febd4e3d ]---
[    1.376873] Kernel panic - not syncing: Fatal exception in interrupt
[    1.383414] ------------[ cut here ]------------
[    1.388174] WARNING: CPU: 0 PID: 0 at kernel/smp.c:633 smp_call_function_many_cond+0x438/0x454
[    1.397027] Modules linked in:
[    1.400174] CPU: 0 PID: 0 Comm:  Tainted: G      D           5.10.96 #0
[    1.406974] Stack : 10000f00 00000000 00989680 8008dc18 80830000 80727f5c 00000000 00000000
[    1.415582]         8140b714 81100000 806f6270 81438478 807b8fe7 10000f00 8140b6b8 5535da56
[    1.424184]         00000000 00000000 806f6270 8140b550 00000000 00000000 ffffefff ffffffea
[    1.432786]         000000b2 8140b55c 000000b2 807c2818 00000009 10000f00 00000000 806f0000
[    1.441387]         00000009 81438080 807b9100 00000000 00000858 00000000 80007810 00000084
[    1.449989]         ...
[    1.452499] Call Trace:
[    1.455024] [<8001a230>] show_stack+0x30/0x100
[    1.459590] [<80379878>] dump_stack+0xa8/0xd8
[    1.464090] [<8003aac4>] __warn+0xc0/0x12c
[    1.468280] [<8003ab8c>] warn_slowpath_fmt+0x5c/0xac
[    1.473387] [<800c4344>] smp_call_function_many_cond+0x438/0x454
[    1.479567] [<800c43a4>] smp_call_function+0x24/0x30
[    1.484677] [<8003a530>] panic+0x114/0x31c
[    1.488889] [<8001a4ec>] die+0x124/0x130
[    1.492920] [<800257c0>] do_page_fault+0x534/0x544
[    1.497849] [<8002a694>] tlb_do_page_fault_0+0x124/0x12c
[    1.503320] [<8006b6d4>] account_system_time+0x60/0xa8
[    1.508589] 
[    1.510114] ---[ end trace a5823539febd4e3e ]---
[    1.514876] Rebooting in 1 seconds..
[    2.513973] triggering watchdog soft-reset...<0xff>HELO

```

aparcar committed:
bcm63xx: fix booting with Kernel 5.10

This is a workaround to make the target overall bootable. With this more
people should be able to test the Kernel 5.10 and report further issues.

Suggested-by: Daniel González Cabanelas <dgcbueu@gmail.com>
Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar commented:
@hauke @danitool I added the suggested workaround and would proceed merging this. More people can try it and if to many issues appear, we disable it before the next release.

Noltari commented:
> @hauke @danitool I added the suggested workaround and would proceed merging this. More people can try it and if to many issues appear, we disable it before the next release.

@aparcar LGTM, you can merge this

aparcar commented:
Merged.
